File: restaurantservice/supply-service/src/main/java/pl/restaurant/supplyservice/api/response/GoodView.java
package pl.restaurant.supplyservice.api.response;

import lombok.*;
import pl.restaurant.supplyservice.api.request.TaxType;

import java.math.BigDecimal;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class GoodView {
    private Long id;
    private Integer ingredientId;
    private BigDecimal quantity;
    private Integer unitId;
    private BigDecimal unitNetPrice;
    private BigDecimal discount;
    private BigDecimal netPrice;
    private TaxType taxType;
    private BigDecimal taxPrice;
}

File: restaurant-app/src/app/model/invoice/good-view.ts
import {TaxType} from "./tax-type";

export interface GoodView {
  id: number;
  ingredientId: number;
  ingredient: string | any;
  quantity: number;
  unitId: number | any;
  unitNetPrice: number;
  discount: number;
  netPrice: number;
  taxType: TaxType | any;
  taxPrice: number;
}
